collect_app/src/main/java/org/odk/collect/android/forms/DatabaseFormsRepository.java

```diff
@@ -40,16 +40,16 @@ public boolean contains(String jrFormId) {
     @Nullable
     @Override
     public Form get(Long id) {
-        try (Cursor cursor = new FormsDao().getFormsCursor(_ID + "=?", new String[]{id.toString()})) {
-            return getFormOrNull(cursor);
-        }
+        return queryForForm(_ID + "=?", new String[]{id.toString()});
     }
 
     @Nullable
     @Override
-    public Form get(String jrFormId, String jrVersion) {
-        try (Cursor cursor = new FormsDao().getFormsCursor(JR_FORM_ID + "=? AND " + JR_VERSION + "=?", new String[]{jrFormId, jrVersion})) {
-            return getFormOrNull(cursor);
+    public Form get(String jrFormId, @Nullable String jrVersion) {
+        if (jrVersion != null) {
```

Comment:
Feels too bad to have a mix of raw queries passed in to the dao layer and higher-level calls on the dao. Wouldn't `getFormsCursorSortedByDateDesc` do what you need here?

Reply:
You're right we should be clearer around which way we're going here: Are we wrapping the Dao or trying to remove it and wrapping the content provider? Personally I'm leaning towards the latter.

To your comment below (as it's related and I can't reply for some reason): I think I'd like to have testing around the repositories as I would envision us being able to create a new one (using Room for example) and we'd be able to just add another `FormRepositoryTest` to drive that out. Right now the actual test is completely divorced from the querying/storage. 

Maybe it makes sense to follow up with a PR trying to inline as much of the Dao as possible into the repo? That's conditional on us agreeing that's the direction we want to go of course!

Reply:
Ah yes, right, I forgot that we have too many layers going on. I agree that a goal would be to remove the existing DAO layer. I think that for now making the repositories use the content providers makes sense. If we were to switch to something like using Room, I think we'd flip that around. But doesn't matter for now.